### PR TITLE
Fix globalThis error

### LINKED ISSUE #4350: iOS - Can't find variable: globalThis
### Description

Using `10.20.0-alpha.2` (but without actually using Hermes on iOS), reported the above crash for a user right on app startup.

### Stacktrace & log output

```shell
ReferenceError
Can't find variable: globalThis

    node_modules/realm-network-transport/dist/bundle.react-native.es.js:91:43 
    node_modules/metro-runtime/src/polyfills/require.js:340:6 loadModuleImplementation
    node_modules/realm/lib/extensions.js:60:45 exports
    node_modules/realm/lib/react-native.js:45:24 
    node_modules/metro-runtime/src/polyfills/require.js:340:6 loadModuleImplementation
    src/db/index.js:8 
    node_modules/metro-runtime/src/polyfills/require.js:340:6 loadModuleImplementation
    src/App.js:31 
    node_modules/metro-runtime/src/polyfills/require.js:340:6 loadModuleImplementation
    node_modules/metro-runtime/src/polyfills/require.js:200:45 guardedLoadModule
    index.js:86:29 
    node_modules/regenerator-runtime/runtime.js:63:44 tryCatch
    node_modules/regenerator-runtime/runtime.js:294:30 
    node_modules/regenerator-runtime/runtime.js:63:44 tryCatch
    node_modules/regenerator-runtime/runtime.js:155:28 invoke
    node_modules/regenerator-runtime/runtime.js:165:19 
    node_modules/promise/setimmediate/core.js:37:14 tryCallOne
    node_modules/promise/setimmediate/core.js:123:25 
    node_modules/react-native/Libraries/Core/Timers/JSTimers.js:248:17 
    node_modules/react-native/Libraries/Core/Timers/JSTimers.js:112:6 _callTimer
    node_modules/react-native/Libraries/Core/Timers/JSTimers.js:166:15 _callReactNativeMicrotasksPass
    node_modules/react-native/Libraries/Core/Timers/JSTimers.js:418:11 callReactNativeMicrotasks
    node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:393:11 value
    node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:135:11 
    node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:370:8 value
    node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:134:17 value
    [native code] value
    [native code] value
```
```


### Can you reproduce a bug?

Not yet

### Reproduction Steps

Regular offline (non sync) usage.

### Version

10.20.0-alpha.2

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

_No response_

### Platform OS and version(s)

iOS 12.1.1 (iPad Mini 3)

### PR DESCRIPTION
**Pull request by takameyer**
## What, How & Why?
`globalThis` was added to the JSC in iOS 12.2.  This causes errors in older devices that aren't being offered updates anymore.  The change here will offer an alternative if `globalThis` is undefined.

This closes #4350

## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 🚦 Tests
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ x.x.x Release notes (yyyy-MM-dd)
 ### Fixed
 * <How to hit and notice issue? what was the impact?> ([#????](https://github.com/realm/realm-js/issues/????), since v?.?.?)
 * Fixed issue that could cause mangling of binary data on a roundtrip to/from the database ([#4278](https://github.com/realm/realm-js/issues/4278), since v10.1.4).
+* Fixed `globalThis` undefined issue for older devices([#4390](https://github.com/realm/realm-js/4390))
```
Comment (kneth) [inline CHANGELOG.md:9]:
```suggestion
* Fixed `globalThis` undefined issue for older devices. ([#4390](https://github.com/realm/realm-js/4390))
```

**Comment by kraenhansen** [comment]
For what it's worth: I approve this change 👍

**Comment by dsy401** [comment]
When will it be released?

**Comment by takameyer** [comment]
Very soon.

**Comment by takameyer** [comment]
@sunainag-c-hrbl It should be, yes.  The code now just checks if `globalThis` exists, and if it doesn't it uses an alternative method.  If `globalThis` is not available on Android, it is possible to update the JSC engine to make sure it does. 
This project can help with that: https://github.com/react-native-community/jsc-android-buildscripts

If you are seeing this on Android, please make a new issue an include which version of React Native and Realm you are using.  Then we can have visibility on that and propose a solution for others having the same issue.

**Comment by takameyer** [comment]
@Yupeng-li Sorry about the oversight!  We will get that fixed and make a new release soon.

**Comment by liamjones** [comment]
@cristianoccazinsp 10.15.0 was the version which will force `realm-network-transport` to the fixed version (0.7.2).

Note that any version since 10.12.0 has the version specified for this package as `^0.7.1` so as long as you upgrade `realm-network-transport` package to 0.7.2 you should have this fix.

**Pull request merged**